src/components/molecules/Visualizer/Widget/index.tsx

```diff
@@ -25,17 +31,25 @@ export default function WidgetComponent<PP = any, SP = any>({
   ...props
 }: Props<PP, SP>) {
   const { align, location } = widgetLayout ?? {};
-  const w = useMemo<Widget | undefined>(
+  const w = useMemo<RawWidget | undefined>(
     () =>
       widget
         ? {
             ...widget,
+            extended: !widgetLayout
+              ? undefined
+              : {
+                  horizontally: !!widget.extended && isHorizontal(widgetLayout.location),
+                  vertically: !!widget.extended && isVertical(widgetLayout.location),
```

Comment:
Please correct me if I am wrong, but this code looks like it is setting horizontally or vertically to true depending on the Widget's current location (widgetLayout.location), right?
What we need the Widget component to know is if it (from the manifest) is **allowed** to be horizontal, vertical, both, or neither. Doesn't this code just **tell** the widget that it is allowed depending on its location and if it is extended?

Reply:
It is the backend's responsibility to restrict the extended, not the plugin's. Yes, and that information is enough for the plugin to set the style.